src/module.c

```diff
@@ -12350,6 +12350,7 @@ int moduleLoad(const char *path, void **module_argv, int module_argc, int is_loa
     if (post_load_err) {
         moduleUnload(ctx.module->name, NULL);
         moduleFreeContext(&ctx);
+        dlclose(handle);
```

Comment:
It looks like moduleUnload handles dlclose today, so I think this will have already been closed.